src/rhsmlib/dbus/objects/register.py

```diff
@@ -262,6 +264,23 @@ def _remove_enable_content_option(options: dict) -> bool:
             enable_content = False
         return enable_content
 
+    def _unregister_and_clean_cp_provider(self, connection_options: dict) -> None:
```

Comment:
I'd probably use a shorter name, eg `_unregister()`: this way, all the bits needed to unregister can be added here, without the need to update the name accordingly in case of changes